frontend/views/containers/chatroom/SendArea.vue

```diff
@@ -190,9 +243,76 @@ export default ({
         return this.createNewLine()
       }
     },
+    updateMentionKeyword () {
+      let value = this.$refs.textarea.value.slice(0, this.$refs.textarea.selectionStart)
+      const lastIndex = value.lastIndexOf('@')
+      const regExWordStart = /(\s)/g // RegEx Metacharacter \s
+      if (lastIndex === -1 || (lastIndex > 0 && !regExWordStart.test(value[lastIndex - 1]))) {
+        return this.endMention()
+      }
+      value = value.slice(lastIndex + 1)
+      if (regExWordStart.test(value)) {
+        return this.endMention()
+      }
+      this.startMention(value, lastIndex)
+    },
+    handleKeydown (e) {
+      if (Object.values(caretKeyCodes).includes(e.keyCode)) {
+        const nChoices = this.ephemeral.mention.options.length
+        if (nChoices &&
+          (e.keyCode === caretKeyCodes.ArrowUp || e.keyCode === caretKeyCodes.ArrowDown)) {
+          const offset = e.keyCode === caretKeyCodes.ArrowUp ? -1 : 1
+          const newIndex = (this.ephemeral.mention.index + offset + nChoices) % nChoices
+          this.ephemeral.mention.index = newIndex
+
+          const { clientHeight, scrollHeight } = this.$refs.mentionWrapper
+          if (scrollHeight !== clientHeight) {
+            const offsetTop = this.$refs.mention[newIndex].offsetTop + this.$refs.mention[newIndex].clientHeight
+
+            this.$refs.mentionWrapper.scrollTo({
+              left: 0, top: Math.max(0, offsetTop - clientHeight)
+            })
+          }
+
+          e.preventDefault()
+        } else {
+          this.endMention()
+        }
+      }
+    },
+    handleKeyDownEnter () {
+      if (this.ephemeral.mention.options.length) {
+        this.addSelectedMention(this.ephemeral.mention.index)
+      } else {
+        this.sendMessage()
+      }
+    },
+    handleKeyDownTab (e) {
+      if (this.ephemeral.mention.options.length) {
+        this.addSelectedMention(this.ephemeral.mention.index)
+        e.preventDefault()
+      }
+    },
     handleKeyup (e) {
       if (e.keyCode === 13) e.preventDefault()
       else this.updateTextArea()
+
+      if (!Object.values(caretKeyCodes).includes(e.keyCode) &&
+        !Object.values(functionalKeyCodes).includes(e.keyCode)) {
```

Comment:
Same remark here as in `handleKeydown()`.

Reply:
Same as above.